src/Lykke.Job.BlockchainBalancesReport/Clients/Steemit/SteemetDeserializer.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Newtonsoft.Json;
+
+namespace Lykke.Job.BlockchainBalancesReport.Clients.Steemit
+{
+    public static class SteemetDeserializer
+    {
+        public static IEnumerable<(string txId, DateTime timestamp, string from, string to, decimal amount)> DeserializeTransactionsResp(string source)
+        {
+            var model = JsonConvert.DeserializeObject<TransactionsResp>(source);
+            var txs = model.Result.Select(p => p[1]).ToList();
+
+            var ids = new HashSet<string>();
+
+            foreach (var tx in txs)
+            {
+                var t = JsonConvert.SerializeObject(tx.op);
+                var t2 = JsonConvert.DeserializeObject<dynamic[]>(t);
+                var op = t2[1];
+                if (t2[0] == "transfer")
+                {
+                    var id = tx.trx_id.ToString();
+
+                    if (!ids.Contains(id))
+                    {
+                        yield return (
+                            id,
+                            DateTime.Parse(tx.timestamp.ToString()),
+                            op.from.ToString(),
+                            op.to.ToString(),
+                            ParseSteemValue(op.amount.ToString()));
+
+                        ids.Add(id);
+                    }
+                }
+            }
+        }
+
+        private static decimal ParseSteemValue(string sourceValue)
+        {
+            return decimal.Parse(sourceValue.Replace(" STEEM", ""), CultureInfo.InvariantCulture);
```

Comment:
Aside from STEEM there is SBD currency which could be transferred too. This line will throw on such transaction.

Reply:
It's ok for now. We'll fix it once first not STEEM will be found. Atm there are no SBD transactions on our hot wallet